### PR TITLE
Added metrics for observed waypoints

### PR DESCRIPTION
**Pull request by eastorski**

### CODE REVIEW

File: erigon-lib/common/metrics/block_metrics.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ var (
 	BlockConsumerPreExecutionDelay   = metrics.NewSummary(`block_consumer_delay{type="pre_execution"}`)
 	BlockConsumerPostExecutionDelay  = metrics.NewSummary(`block_consumer_delay{type="post_execution"}`)
 	BlockProducerProductionDelay     = metrics.NewSummary(`block_producer_delay{type="production"}`)
+	ObservedWaypointCheckpointLength = metrics.NewSummary(`observed_waypoint_length{type="checkpoint"}`)
```
Comment (taratorio) [inline erigon-lib/common/metrics/block_metrics.go:34]: would suggest to move these inside the `heimdall` package

File: polygon/sync/tip_events.go [hunk 2]
```diff
@@ -201,6 +203,16 @@ func (te *TipEvents) Run(ctx context.Context) error {
 	}, heimdall.WithEventsLimit(5))
 	defer milestoneObserverCancel()
 
+	metricsMilestoneObserverCancel := te.heimdallObserverRegistrar.RegisterMilestoneObserver(func(milestone *heimdall.Milestone) {
```
Comment (taratorio) [inline polygon/sync/tip_events.go:206]: would suggest to move these inside the `heimdall` package, can be registered as background daemons of the `heimdall.Service` inside its `Run` method

File: erigon-lib/common/metrics/block_metrics.go [hunk 3]
```diff
@@ -31,6 +31,8 @@ var (
...
+	ObservedWaypointMilestoneLength  = metrics.NewSummary(`observed_waypoint_length{type="milestone"}`)
```
Comment (taratorio) [inline erigon-lib/common/metrics/block_metrics.go:35]: do we need these to be `summary`-ies or would not a `gauge` metric type be more suitable?

File: polygon/sync/tip_events.go (hunk 2 above)
Comment (eastorski) [reply polygon/sync/tip_events.go]: moved to heimdall

**Review by taratorio — APPROVED** [review]

**Pull request merged**